polymer/src/sample-category-data.html

```diff
@@ -0,0 +1,207 @@
+
+      static get properties() { return {
+
+        categoryName: String,
+
+        itemName: String,
+
+        categories: {
+          type: Array,
+          notify: true
+        },
+
+        category: {
+          type: Object,
+          notify: true
+        },
+
+        item: {
+          type: Object,
+          computed: '_computeItem(category.items, itemName)',
+          notify: true
+        },
+
+        failure: {
+          type: Boolean,
+          notify: true,
+          readOnly: true
+        },
+
+        resource: {
+          type: Object
+        }
+
+      }}
+
+      static get observers() { return [
+        '_categoriesLoaded(categories, categoryName)', '_categoryNameSet(categoryName)'
+      ]}
+
+      static get sitemapUrl() { return 'https://ampbyexample.com/sitemap.json'; }
+
+      _categoriesLoaded(categories, categoryName) {
+        this._computeCategory(categoryName)
+      }
+
+      _categoryNameSet(categoryName) {
+        this._fetchCategories(3);
+      }
+
+      handleResponse(data) {
+        this.resource = data.detail.response[0]["categories"]
+        var catNames =  this.resource.map(function(cat) {
+          var obj = {
+            name:  cat["name"],
+            id: cat["id"]
+          };
+          return obj;
+        });
+        this.categories = catNames;
+      }
+
+      _getCategoryObject(categoryName, categories) {
+        if (categories === undefined) {
+          return null;
+        }
+        for (let i = 0, c; c = categories[i]; ++i) {
+          if (c.id === categoryName) {
+            this.category = c;
+            return c;
+          } else if (categoryName === "index.html") {
+            return null
+          }
+        }
+      }
+
+      _computeCategory(categoryName) {
+
+        // Fetch the items of the category. Note that the fetch is asynchronous,
+        // which means `category.items` may not be set initially (but that path
+        // will be notified when the fetch completes).
+        let categoryObj = this._getCategoryObject(categoryName, this.categories);
+        this._fetchItems(categoryObj, 1);
+        return categoryObj;
+      }
+
+      _computeItem(items, itemName) {
+        if (!items || !itemName) {
+          return;
+        }
+        for (let i = 0, item; item = items[i]; ++i) {
+          if (item.url.endsWith(itemName.concat('/'))) {
+            return item;
+          }
+        }
+      }
+
+      _parseCategoriesFromJson(e) {
+        return JSON.parse(e.target.responseText)[0]["categories"]
+      }
+
+      _fetchCategories(attempts) {
+        this._setFailure(false);
+        if (this.categories) {
+          return;
+        }
+        this._getResource({
+          url: SampleCategoryData.sitemapUrl,
+          onLoad(e) {
+            this.resource = this._parseCategoriesFromJson(e)
+            var catNames =  this.resource.map(function(cat) {
+              var obj = {
+                name:  cat["name"],
```

Comment:
nit: you're mixing double quotes and single quotes, it's better to be consistent.

Reply:
Done